MuvrKit/MKWatchConnectivity.swift

```diff
@@ -249,20 +249,17 @@ public final class MKConnectivity : NSObject, WCSessionDelegate {
 
             // update the number of recorded samples
             let readFromDate = NSDate(timeIntervalSince1970: sensorData.end)
-            let updatedProps = props.with(accelerometerEnd: readFromDate)
-            self.sessions[session] = updatedProps
+            if let p = self.sessions[session] {
+                self.sessions[session] = p.with(accelerometerEnd: readFromDate)
+            }
             
             // transfer what we have so far
-            transferSensorDataBatch(sensorData, session: session, props: updatedProps) {
-                // REVIEW: does this fix the sessions problem?
-                if self.sessions[session] == nil {
-                    NSLog("Session \(session) already removed.")
-                    dispatch_async(self.transferQueue, processFirstSession)
-                } else {
-                    // set the expected range of samples on the next call
-                    let finalProps = updatedProps.with(accelerometerStart: readFromDate)
```

Comment:
updatedProps was read before sending the data and might be out of date at this point.